apps/backend/src/libs/modules/server-application/base-server-application.ts

```diff
@@ -166,16 +166,19 @@ class BaseServerApplication implements ServerApplication {
 	}
 
 	public addRoute(parameters: ServerApplicationRouteParameters): void {
-		const { handler, method, path, validation } = parameters;
+		const { handler, method, path, preHandler, validation } = parameters;
 
-		this.app.route({
+		const routeOptions = {
 			handler,
 			method,
 			schema: {
 				body: validation?.body,
 			},
 			url: path,
-		});
+			...(preHandler ? { preHandler } : {}),
```

Comment:
now it's array so we can do it simpler
```suggestion
			preHandler: preHandlers ?? [],
```